src/Persistence/Repository/ResetPasswordRequestRepositoryTrait.php

```diff
@@ -61,8 +61,13 @@ public function getMostRecentNonExpiredRequestDate(object $user): ?\DateTimeInte
 
     public function removeResetPasswordRequest(ResetPasswordRequestInterface $resetPasswordRequest): void
     {
-        $this->getEntityManager()->remove($resetPasswordRequest);
-        $this->getEntityManager()->flush();
+        $this->createQueryBuilder('t')
+            ->delete()
+            ->where('t.user = :user')
+            ->setParameter('user', $resetPasswordRequest->getUser())
+            ->getQuery()
+            ->execute()
+        ;
```

Comment:
Clever way to not break BC but accomplish the job.

But we need to clarify the guidance above the interface method: https://github.com/SymfonyCasts/reset-password-bundle/blob/master/src/Persistence/ResetPasswordRequestRepositoryInterface.php#L58

Probably:

> Remove this reset password from persistence and any other for this user.
>
> This method should remove *this* ResetPasswordRequestInterface and *also*
> all other ResetPasswordRequestInterface objects in storage for the *same* user.

Reply:
Yeah, the BC break is only that they previously weren't all deleted and now they are.

Reply:
I'm ok with that - it's a behavior change for security purposes